example/canny.cpp

```diff
@@ -0,0 +1,165 @@
+#include <boost/gil.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/extension/numeric/sampler.hpp>
+#include <boost/gil/extension/numeric/resample.hpp>
+#include <boost/gil/extension/numeric/kernel.hpp>
+#include <boost/gil/extension/numeric/convolve.hpp>
+
+#include <math.h>
+
+using namespace std;
+using namespace boost::gil;
+
+//Example how to detect edges via Canny algorithm
+
+template <typename SrcView>
+void smoothing(const SrcView &src)
+{
+    float gaussian[] = {0.06136f, 0.24477f, 0.38774f, 0.24477f, 0.06136f};
+    kernel_1d<float> kernel(gaussian, 5, 2);
+
+    convolve_rows<gray32f_pixel_t>(src, kernel, src, convolve_option_output_ignore);
+    convolve_cols<gray32f_pixel_t>(src, kernel, src, convolve_option_output_ignore);
+}
+
+void sobel_filtering(const gray8c_view_t &src,
+                     const gray8_view_t &magnitude, const gray32f_view_t &slope)
+{
+    float first_sobel[] = {1.f, 0.f, -1.f};
+    float second_sobel[] = {1.f, 2.f, 1.f};
+
+    kernel_1d<float> first_sobel_kernel(first_sobel, 3, 1);
+    kernel_1d<float> second_sobel_kernel(second_sobel, 3, 1);
+
+    gray16s_image_t vertical(src.dimensions());
+    gray16s_image_t horizontal(src.dimensions());
+
+    convolve_rows<gray32f_pixel_t>(src, first_sobel_kernel,
+                                   view(vertical), convolve_option_output_zero);
+    convolve_cols<gray32f_pixel_t>(const_view(vertical), second_sobel_kernel,
+                                   view(vertical), convolve_option_output_zero);
+
+    convolve_rows<gray32f_pixel_t>(src, second_sobel_kernel, view(horizontal),
+                                   convolve_option_output_zero);
+    convolve_cols<gray32f_pixel_t>(const_view(horizontal), first_sobel_kernel,
+                                   view(horizontal), convolve_option_output_zero);
+
+    //Magnitude and angle
+    auto ver_it = view(vertical).begin();
+    auto hor_it = view(horizontal).begin();
+    auto slope_it = slope.begin();
+
+    for (auto mag_it = magnitude.begin(); mag_it != magnitude.end();
+            ++mag_it, ++ver_it, ++hor_it, ++slope_it) {
+
+        *mag_it = std::min(255,
+                           (int)std::sqrt(
+                               std::pow((int)(*ver_it), 2) + std::pow((int)(*hor_it), 2)));
+
+        *slope_it = (float)std::atan2((int)(*hor_it), (int)(*ver_it));
+    }
+}
+
+void suppression_threshold(const gray8c_view_t &magnitude, const gray32fc_view_t &angle,
```

Comment:
`non_maximal_suppression` wouldn't it be a better name?

Reply:
This function makes suppression and threshold. Full name should be like `non_maximal_suppression_and_thresholding`, but it is too long, isn't it?

Reply:
But, isn't the thresholding a technique, a mean, to perform the suppression?

I wouldn't worry about too long names. This is an example, it should be clear, self-descriptive, raise no questions in someone who reads the code and expects to learn from the code.